metals/src/main/scala/scala/meta/internal/metals/watcher/SimpleFileWatcher.scala

```diff
@@ -0,0 +1,66 @@
+package scala.meta.internal.metals.watcher
+
+import collection.JavaConverters._
+import java.nio.file.{FileSystems, Path, StandardWatchEventKinds}
+import java.util.concurrent.BlockingQueue
+import scala.concurrent.{ExecutionContext, Future}
+
+object SimpleFileWatcher {
+  /*
+   * returns a function which closes the file watcher when called
+   */
+  def initWatcher(
+    watchFilter: Path => Boolean,
+    watchEventQueue: BlockingQueue[FileWatcherEvent],
+    filesToWatch: Set[Path],
+    directoriesToWatch: Set[Path],
+  )(implicit ec: ExecutionContext): () => Unit = {
+    scribe.error(s"RFERGUSON INIT WATCHER")
+    val watchService = FileSystems.getDefault().newWatchService();
+    filesToWatch.filter(watchFilter).foreach { path =>
```

Comment:
If `directoriesToWatch` the set of parent directories of `filesToWatch` have any overlap, do you know if multiple watches will be registered? It might be better to normalize both sets' paths, take the union between them, and register file watches on those paths.

That could also help to reduce the redundancy here.